app/models/manageiq/providers/autosde/storage_manager/physical_storage.rb

```diff
@@ -37,21 +37,31 @@ def raw_update_physical_storage(options = {})
 
   def self.raw_create_physical_storage(ext_management_system, options = {})
     sys_to_create = ext_management_system.autosde_client.StorageSystemCreate(
-      :password       => options['password'],
-      :user           => options['user'],
-      :system_type    => PhysicalStorageFamily.find(options['physical_storage_family_id']).name,
-      :management_ip  => options['management_ip'],
+      :password => options['password'],
+      :user => options['user'],
+      :system_type => PhysicalStorageFamily.find(options['physical_storage_family_id']).name,
+      :management_ip => options['management_ip'],
       :storage_family => "ontap_7mode"
     )
 
     begin
       new_storage = ext_management_system.autosde_client.StorageSystemApi.storage_systems_post(sys_to_create)
+      ### SETUP SERVICE CATALOG COMPONENTS ###
+      # CATALOG
+      catalog_id = ServiceTemplateCatalog.find_or_create_by(:name => "AutoSDE Catalog").id
+      # DIALOG
+      vol_dialog = YAML.load_file('plugins/manageiq-providers-autosde/service_catalog/volume/dialog.yaml')
+      vol_dialog_id = DialogImportService.new.import(vol_dialog).id
+      # ITEM
+      vol_item_prop = { :name => "Volume", :prov_type => "autosde", :display => "true", :service_template_catalog_id => catalog_id,
+                        :config_info => { :provision => { :fqname => "/Storage/AutoSde/Services/ServiceVolumeRequestApproval/Default", :dialog_id => vol_dialog_id } } }
+      ServiceTemplate.create_catalog_item(vol_item_prop, nil) unless ServiceTemplate.find_by(:name => vol_item_prop[:name]) != nil
```

Comment:
Is this just for your dev/testing?  If not then I don't think this is the right place to do this

Reply:
We want to create for autosde clients a service catalog item that will automatically create a volume with the appropriate dialog.
The catalog will be available after connecting to physical storage.
Where should we put this script? in another function? or another file?

Reply:
I put it in another file.

Reply:
Setting up a service catalog should be done by the user not automatically in code like this.